k2/python/k2/fsa.py

```diff
@@ -226,14 +224,14 @@ def __setattr__(self, name: str, value: Any) -> None:
         '''
 
         if hasattr(type(self), name) or name in self.__dict__:
-            # For attribute names that have a getter/setter function (like 'property'
-            # or which exist as members of the object's __dict__ (like
-            # 'arcs', '_tensor_attr', '_non_tensor_attr'), do the member lookup
-            # in the normal way as if __setattr__ was not defined.
-            object.__setattr__(key, value)
-
-        if name in ('_tensor_attr', '_non_tensor_attr', 'arcs',
-                    '_cache'):
+            # For attribute names that have a getter/setter function
+            # (like 'property' or which exist as members of the object's
+            # __dict__ (like 'arcs', '_tensor_attr', '_non_tensor_attr'),
+            # do the member lookup in the normal way as if __setattr__ was
+            # not defined.
+            object.__setattr__(name, value)
```

Comment:
object.__setattr__(self, name, value)

Reply:
mm, you're right, this code was doubly wrong.  Also it should have  had a `return` statement afterwards, according to the original intention of the code.  But I now realize that the whole thing is unnecessary, and better to delete it.

Reply:
Let me rework this slightly.

Reply:
Fixing this via separate PR

Reply:
Mark as resolved. This pullrequest has been rebased to the latest master.